example_client/grpc_stateful_client.py

```diff
@@ -32,154 +32,250 @@ def printDebug(msg):
+            if x == 0:
+                request.inputs['sequence_control_input'].CopyFrom(make_tensor_proto(SEQUENCE_START, dtype="uint32"))
+        
+            # Set sequence id
+            request.inputs['sequence_id'].CopyFrom(make_tensor_proto(sequence_id, dtype="uint64"))
 
-    samples_limit = int(args.get('samples'))
-    for x in range(0, batch_size):
+            # Add sequence end
+            if x == batch_size + cw_l + cw_r - 1:
+                request.inputs['sequence_control_input'].CopyFrom(make_tensor_proto(SEQUENCE_END, dtype="uint32"))
 
-        if x >= samples_limit:
-            break
+            start_time = datetime.datetime.now()
+            result = stub.Predict(request, 10.0) # result includes a dictionary with all model outputs
+            end_time = datetime.datetime.now()
 
-        printDebug('\tExecution: {}\n'.format(x))
-        request = predict_pb2.PredictRequest()
-        request.model_spec.name = args.get('model_name')
+            ValidateOutput(result, output_names)
 
-        printDebug('\tTensor before input in shape: {}\n'.format(obj[x].shape))
-        inputArray = np.expand_dims(obj[x], axis=0)
-        printDebug('\tTensor input in shape: {}\n'.format(inputArray.shape))
+            duration = (end_time - start_time).total_seconds() * 1000
+            processing_times = np.append(processing_times,np.array([int(duration)]))
 
-        request.inputs[args['input_name']].CopyFrom(make_tensor_proto(inputArray, shape=inputArray.shape))
+            # Compare results after we are pass initial context window results
+            if score_index >= 0:
+                # Loop over reference output results
+                errPerNameSum = 0.0
 
-        if x == 0:
-            request.inputs['sequence_control_input'].CopyFrom(make_tensor_proto(SEQUENCE_START, dtype="uint32"))
-        
-        request.inputs['sequence_id'].CopyFrom(make_tensor_proto(sequence_id, dtype="uint64"))
+                for output_name in output_names:
+                    scoreData = referenceArrays[output_name][key][score_index]
 
-        if x == batch_size - 1 or x == samples_limit - 1:
-            request.inputs['sequence_control_input'].CopyFrom(make_tensor_proto(SEQUENCE_END, dtype="uint32"))
+                    # Parse output
+                    resultsArray = make_ndarray(result.outputs[output_name])
 
-        start_time = datetime.datetime.now()
-        result = stub.Predict(request, 10.0) # result includes a dictionary with all model outputs
-        end_time = datetime.datetime.now()
-        if args['output_name'] not in result.outputs:
-            print("ERROR: Invalid output name", args['output_name'])
-            print("Available outputs:")
-            for Y in result.outputs:
-                print(Y)
-            exit(1)
+                    # Calculate error
+                    meanErr = CalculateUtteranceError(scoreData, resultsArray[0])
+                    errPerNameSum += meanErr
```

Comment:
To make it more clear please change `meanErr` to `avg_rms_error` as what we are calculating here is Average Root Mean Square Error.